Minetest 5.0 build fail on Clang

##### Issue type
 - Build issue
 
##### Minetest version
```
5.0.0
```

##### OS / Hardware
Build on OpenMandriva Lx4 (cooker) clang 7.0.1

##### Summary
Along with version Minetest 5.0.0 there was a problem with building it with Clang. 
Previous branch 0.4.X build fine with Clang here but version Minetest 5.0 it is not built correctly.
OpenMandriva builds its packages with Clang. That's why I'm trying to make it possible to build Minetest from Clang.

I would like to add that as a temporary solution, I have correctly built it with GCC. However, this is the only workaround, not a solution.

Build error:
```
In file included from /builddir/build/BUILD/minetest-5.0.0/src/irrlicht_changes/CGUITTFont.h:37:
/builddir/build/BUILD/minetest-5.0.0/src/irrlicht_changes/irrUString.h:2919:22: error: reinterpret_cast from 'const char16_t *' to 'irr::uchar8_t *' (aka 'char *') casts away qualifiers
uchar8_t* ptr8 = reinterpret_cast<uchar8_t*>(ret.c_str());

/builddir/build/BUILD/minetest-5.0.0/src/irrlicht_changes/irrUString.h:2925:22: error: reinterpret_cast from 'const char16_t *' to 'irr::uchar8_t *' (aka 'char *') casts away qualifiers
uchar8_t* ptr8 = reinterpret_cast<uchar8_t*>(ret.c_str());

/builddir/build/BUILD/minetest-5.0.0/src/irrlicht_changes/irrUString.h:2934:14: error: cannot initialize a variable of type 'char16_t *' with an rvalue of type 'const char16_t *'
char16_t* ptr = ret.c_str();
```

Full build log here: http://file-store.openmandriva.org/api/v1/file_stores/1239cc77e197c14934076df64b518f00c7c8cdd3.log?show=true

##### Steps to reproduce
1. Try build with Clang 7.0.1
2. See at build error.

--- a/src/irrlicht_changes/irrUString.h
+++ b/src/irrlicht_changes/irrUString.h
@@ -2916,13 +2916,13 @@ public:
 				ret[0] = unicode::BOM;
 			else if (endian == unicode::EUTFEE_LITTLE)
 			{
-				uchar8_t* ptr8 = reinterpret_cast<uchar8_t*>(ret.c_str());
+				uchar8_t* ptr8 = reinterpret_cast<uchar8_t*>(&ret[0]);
 				*ptr8++ = unicode::BOM_ENCODE_UTF16_LE[0];
 				*ptr8 = unicode::BOM_ENCODE_UTF16_LE[1];
 			}
 			else
 			{
-				uchar8_t* ptr8 = reinterpret_cast<uchar8_t*>(ret.c_str());
+				uchar8_t* ptr8 = reinterpret_cast<uchar8_t*>(&ret[0]);
 				*ptr8++ = unicode::BOM_ENCODE_UTF16_BE[0];
 				*ptr8 = unicode::BOM_ENCODE_UTF16_BE[1];
 			}

